semantics/src/main/scala/aqua/semantics/rules/types/TypesInterpreter.scala

```diff
@@ -333,10 +333,12 @@ class TypesInterpreter[S[_], X](using
   override def ensureTypeMatches(
     token: Token[S],
     expected: Type,
-    givenType: Type
+    givenType: Type,
+    alias: Option[String] = None
```

Comment:
I don't like adding this argument here. `TypesAlgebra` is the one that should know about type aliases, it should not get this info from outside, should it?
Maybe lets add a separate method `resolveNamedTypeConstruction`, which will take named type token and field types, then resolve the type and report a error if the type is not `data` or `ability`?